src/formula/function.cpp

```diff
@@ -701,6 +725,39 @@ DEFINE_WFL_FUNCTION(lerp, 3, 3)
 	return variant(static_cast<int>((lo + alpha * (hi - lo)) * 1000.0), variant::DECIMAL_VARIANT);
 }
 
+DEFINE_WFL_FUNCTION(lerp_index, 2, 2)
+{
+	const std::vector<variant> items = args()[0]->evaluate(variables, fdb).as_list();
+	const double alpha = args()[1]->evaluate(variables, fdb).as_decimal() / 1000.0;
+	// Same formula as red_to_green etc
+	const double val_scaled = std::clamp(0.01 * alpha, 0.0, 1.0);
+	const int idx = int(std::nearbyint((items.size() - 1) * val_scaled));
```

Comment:
Missing a check that items.size() != zero.

Reply:
I added the check to the Lua version of this function too.